lib/dereference.js

```diff
@@ -94,14 +99,37 @@ function crawl (obj, path, pathFromRoot, parents, $refs, options) {
  * @param {string} path - The full path of `$ref`, possibly with a JSON Pointer in the hash
  * @param {string} pathFromRoot - The path of `$ref` from the schema root
  * @param {object[]} parents - An array of the parent objects that have already been dereferenced
+ * @param {object[]} processedObjects - An array of all the objects that have already been dereferenced
```

Comment:
Missing jsdoc annotation for `dereferencedCache`
```suggestion
 * @param {object[]} processedObjects - An array of all the objects that have already been dereferenced
 * @param {object} dereferencedCache - An map of all the dereferenced objects 
```